### PR TITLE
fix: allow closing all REST and GraphQL tabs

### LINKED ISSUE #5497: [feature]: Why must a request tab be kept open?  and cannot be closed
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Summary

Why must a request tab be kept open?  and cannot be closed？This interaction is strange

Image

### Why should this be worked on?

I just switched from Postman, and this interaction feels a bit strange.

### PR DESCRIPTION
**Pull request by sneycampos**
Closes #5497 

### What's changed

- [x] Allow closing the final REST and GraphQL request tab.
- [x] Support zero open tabs in the shared tab service, including nullable active tab state.
- [x] Persist intentionally empty REST and GraphQL tab states across reloads.
- [x] Add a minimal empty workspace state with the Hoppscotch logo, **New Request**, and **Import Collection** actions.
- [x] Keep recently closed tabs working from the empty state.
- [x] Guard REST/GraphQL active-tab consumers so actions no-op safely when no tab is active.
- [x] Ensure REST URL/query params create a request tab before applying params when the workspace is empty.
- [x] Add focused tab service and persistence coverage for empty-tab behavior.

### Notes to reviewers

- Scope is REST and GraphQL tabs only; realtime tabs are unchanged.
- First load without persisted tab state still creates the existing starter tab. The empty state is only restored after the user intentionally closes all tabs.
- The empty-state copy uses i18n keys, with English strings added in this PR.
- Validated with focused typecheck, tab/persistence tests, lint, and manual browser checks for closing tabs, dirty-tab confirmation, empty state actions, and reload persistence.

---
## Summary by cubic
You can now close the last REST or GraphQL tab. When all tabs are closed, an empty workspace appears with New Request and Import Collection, and this zero‑tab state is restored on reload.

- New Features
  - Allow closing the final REST and GraphQL tab; “Close others” is shown only when multiple tabs exist.
  - Add `AppEmptyTabs` for REST and GraphQL with New Request and Import Collection; support reopening recently closed tabs from empty state.
  - Persist zero‑tab state (`lastActiveTabID: null`) and restore it on reload; add i18n copy for the empty state.

- Bug Fixes
  - Preserve OAuth 2.0 tokens on redirect when no tabs are open or after restoring an empty state; create a new REST/GQL tab seeded with the token if needed.
  - Make active‑tab usage null‑safe across the app (REST/GraphQL UIs, codegen, OAuth, environment inputs/tooltips, inspectors, context menus, spotlight, GraphQL connection/subscriptions); guard tab actions when no tab is active.
  - Ensure REST URL/query params open a request tab before applying when the workspace is empty.
  - cURL import: no‑op on example/test‑runner tabs and preserve the current request name.
  - Tab service: `currentTabID` is nullable; closing the active tab activates the nearest remaining tab or clears selection. Update persistence schemas/tests to accept and load `lastActiveTabID: null`; add tests for empty‑state reopen and navigation (including MRU).

<sup>Written for commit ac9cffc1cf02901af37ec8ff1f1b3eb27db16d96. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/hoppscotch/hoppscotch/pull/6286?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by sneycampos** [comment]
@liyasthomas could you take a look? thank you!

File: packages/hoppscotch-common/src/components/graphql/RequestOptions.vue [hunk 1]
```diff
@@ -149,8 +150,8 @@ const runQuery = async (
       url: runURL,
       request: request.value,
       inheritedHeaders,
-      inheritedAuth: tabs.currentActiveTab.value.document.inheritedProperties
-        ?.auth.inheritedAuth as HoppGQLAuth | undefined,
+      inheritedAuth: activeTab?.document.inheritedProperties?.auth
+        .inheritedAuth as HoppGQLAuth | undefined,
```
[the remark this answers is not in the document]
Comment (sneycampos) [reply packages/hoppscotch-common/src/components/graphql/RequestOptions.vue]: done.

File: packages/hoppscotch-common/src/components/graphql/Request.vue [hunk 2]
```diff
@@ -99,22 +100,24 @@ const onConnectClick = () => {
...
 
   connect({
     url: url.value,
-    request: tabs.currentActiveTab.value.document.request,
+    request: activeTab.document.request,
     inheritedHeaders,
-    inheritedAuth: tabs.currentActiveTab.value.document.inheritedProperties
-      ?.auth.inheritedAuth as HoppGQLAuth,
+    inheritedAuth: activeTab.document.inheritedProperties?.auth
+      .inheritedAuth as HoppGQLAuth,
```
[the remark this answers is not in the document]
Comment (sneycampos) [reply packages/hoppscotch-common/src/components/graphql/Request.vue]: done.
